tutorial/03-responding-to-slack-events.md

```diff
@@ -106,18 +94,16 @@ Back to our application, it's time to link our onboarding functionality to Slack
 # ================ Team Join Event =============== #
 # When the user first joins a team, the type of the event will be 'team_join'.
 # Here we'll link the onboarding_message callback to the 'team_join' event.
-@slack_events_adapter.on("team_join")
-def onboarding_message(payload):
+@app.event("team_join")
+def onboarding_message(event):
     """Create and send an onboarding welcome message to new users. Save the
     time stamp of this message so we can update this message in the future.
     """
-    event = payload.get("event", {})
-
     # Get the id of the Slack user associated with the incoming event
     user_id = event.get("user", {}).get("id")
 
     # Open a DM with the new user.
-    response = slack_web_client.conversations_open(users=user_id)
+    response = app.client.conversations_open(users=user_id)
```

Comment:
`app.client` also works in this app as it surely holds the `SLACK_BOT_TOKEN`. Bu, to make this example app reusable for OAuth flow enabled / custom `authorize` enabled apps, we may want to use `client` from listener arguments instead. With the listener arg, we can safely use the client holding a valid token resolved by `authorize` function. 

FYI, refer to https://slack.dev/bolt-python/api-docs/slack_bolt/kwargs_injection/args.html to learn all available listener/middleware arguments.

```suggestion
    response = client.conversations_open(users=user_id)
```

Reply:
@seratch This makes a lot of sense! And thank you for sharing the full reference for the listener arguments available. I've made the updates to use the client passed in as an argument as you've suggested!